reasoner/resolution/framework/ResponseProducer.java

```diff
@@ -75,4 +81,16 @@ public void removeDownstreamProducer(Request request) {
         // note: this is a large performance win when processing large batches of requests
         if (removed) downstreamProducerSelector = downstreamProducer.iterator();
     }
+
+    public int iteration() {
+        return iteration;
+    }
+
+    /**
+     * Prepare a response producer for the another iteration from this one
+     * Notably maintains the set of produced answers for deduplication
+     */
+    public ResponseProducer newIteration(Iterator<ConceptMap> traversalProducer, int iteration) {
```

Comment:
create a new derived `ResponseProducer` from the current one, which saves the set of previously produced answers